java/vector/src/main/java/org/apache/arrow/vector/BaseIntVector.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+/**
+ * Interface for all int type vectors.
+ */
+public interface BaseIntVector extends ValueVector {
+
+  /**
+   * set the encoded value from a {@link org.apache.arrow.vector.dictionary.Dictionary}.
+   */
+  void setEncodedValue(int index, Object value);
```

Comment:
I think it would be better if you made this a long, then down-casted rather then boxing/unboxing.

Reply:
Another alternative might be to use a visitor pattern, but i'm not sure how efficient those can be made in java.

Reply:
My bad, the indices from Dictionary is explicitly int type, Object type is not nedded, fixed now. Thanks!